utils/python/CIME/SystemTests/nck.py

```diff
@@ -1,126 +1,52 @@
-
-    def __init__(self, case):
-        """
-        initialize a test object
-        """
-        SystemTestsCommon.__init__(self, case)
-
-    def build_phase(self, sharedlib_only=False, model_only=False):
-        '''
-        build can be called once (sharedlib_only and model_only both False)
-        or twice (once with each true)
-        This test requires a sharedlib build for both phases
-        we must handle both cases correctly
-        '''
-        exeroot = self._case.get_value("EXEROOT")
-        cime_model = CIME.utils.get_model()
-        if not model_only:
-            machpes1 = os.path.join("LockedFiles","env_mach_pes.orig.xml")
-            if os.path.isfile(machpes1):
-                shutil.copy(machpes1,"env_mach_pes.xml")
-            else:
-                shutil.copy("env_mach_pes.xml", machpes1)
-
-        # Build two exectuables for this test, the first is a default build, the
-        # second halves the number of tasks and runs two instances for each component
-        # Lay all of the components out sequentially
-        for bld in range(1,3):
-            logging.warn("Starting bld %s"%bld)
-            machpes = os.path.join("LockedFiles","env_mach_pes.NCK%s.xml"%bld)
-            if model_only:
-                # This file should have been created in the sharedlib_only phase
-                shutil.copy(machpes,"env_mach_pes.xml")
-                self._case.read_xml()
-            else:
-                for comp in ['ATM','OCN','WAV','GLC','ICE','ROF','LND']:
-                    self._case.set_value("NINST_%s"%comp, bld)
-                    ntasks      = self._case.get_value("NTASKS_%s"%comp)
-                    if(bld == 1):
-                        if ( ntasks > 1 ):
-                            self._case.set_value("NTASKS_%s"%comp, int(ntasks/2))
-                    else:
-                        self._case.set_value("NTASKS_%s"%comp, ntasks*2)
-                    self._case.flush()
-
-            case_setup(self._case, test_mode=True, reset=True)
-            if not sharedlib_only:
-                self.clean_build()
-
-            self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
-            if not model_only:
-                shutil.copy("env_mach_pes.xml", machpes)
-            if not sharedlib_only:
-                shutil.move("%s/%s.exe"%(exeroot,cime_model),"%s/%s.exe.NCK%s"%(exeroot,cime_model,bld))
-                shutil.copy("env_build.xml",os.path.join("LockedFiles","env_build.NCK%s.xml"%bld))
-
-        # Because mira/cetus interprets its run script differently than
-        # other systems we need to copy the original env_mach_pes.xml back
-#        shutil.copy(machpes1,"env_mach_pes.xml")
-#        shutil.copy("env_mach_pes.xml",
-#                    os.path.join("LockedFiles","env_mach_pes.xml"))
+class NCK(SystemTestsCompareTwo):
 
-    def run_phase(self):
-        os.chdir(self._caseroot)
+    _COMPONENT_LIST = ('ATM','OCN','WAV','GLC','ICE','ROF','LND')
```

Comment:
Good question: I wondered the same thing. I just copied what was already being done for the NCK test. I noticed that ESP was absent from this list, and wasn't sure if that was intentional. CPL / DRV is also absent, and I'm pretty sure that _is_ intentional. If you are able to tell me what this is supposed to look like (e.g., whether ESP is supposed to be excluded), then I can change this.